designs/2023-relative-cache/README.md

```diff
@@ -0,0 +1,65 @@
+- Repo: eslint/eslint
+- Start Date: 2023-02-14
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Christian Schulz (@cschulz)
+
+# Relative cache location strategy
+
+## Summary
+
+To be able to share the created eslint cache between different systems f.e. local machine and CI, there is a need for a relative path stored in the cache.
+
+## Motivation
+
+The overall motivation for doing this is performance during the eslint process.
+
+Most people are running eslint on local developer machine and additional on the CI workers.
+In high changing repositories there are several pull requests at the same time changing different files requiring linting all the time.
+This is a time consuming process depending on the enabled rules like type checking rules.
+
+If the CI worker is able to cache its last run, it can just pick this cache up and use it.
+Also developers can reuse this cache or commit it as part of a repository for the CI.
+
+## Detailed Design
+
+The LintResultCache takes a file path as parameter to find or store a file cache entry.
+
+One approach would be to truncate the given absolute path of the file by the path of the cache location.
+
+Another approach to get a relative path is to use the parent eslint config file if we are thinking about config hierarchies.
+
+The configuration should be done by CLI flags like the cache strategy option itself.
+
+## Documentation
+
+Add the new CLI flag to the CLI documentation.
+
+## Drawbacks
+
+No.
+
+## Backwards Compatibility Analysis
```

Comment:
What about implementing this as a breaking change, with relative paths as the default/only supported behavior, so that we don't have to add a new CLI option? Is there any reason someone would still want/need to use caching files with absolute paths? If we were implementing caching from scratch, we wouldn't bother supporting two different styles, right?

Reply:
In my opinion it would be also a possibility. It would be a onetime break for all caches but then they can use the new system.